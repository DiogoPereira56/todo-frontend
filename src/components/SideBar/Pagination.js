import { PropTypes } from 'prop-types'
import { PaginationWrapper, Paginnation } from './SideBar.styles'

const Pagination = ({ listsPerPage, totalLists, setCurrentPage }) => {
    const pageNumbers = [];

    for (let i = 1; i <= Math.ceil(totalLists / listsPerPage); i++){
        pageNumbers.push(i);
    }

    return (
        <nav>
            <PaginationWrapper>
                {pageNumbers.map(number => (
                    <Paginnation key={number} onClick={() => setCurrentPage(number)} >
                        {number}
                    </Paginnation>
                ))}
            </PaginationWrapper>
        </nav>
    )

}

Pagination.propTypes = {
    listsPerPage: PropTypes.number,
    totalLists: PropTypes.number,
    setCurrentPage: PropTypes.func
  };

export default Pagination;