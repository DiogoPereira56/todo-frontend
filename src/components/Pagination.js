import { PropTypes } from 'prop-types';
import { useEffect, useState } from 'react';
import { PaginationWrapper, PaginnationNumber } from './SideBar/SideBar.styles';

const Pagination = ({ listsPerPage, totalLists, setCurrentPage }) => {
    const [pageNumbers, setPageNumbers] = useState([1]);
    const temp = [];

    useEffect(() => {
        for (let i = 1; i <= Math.ceil(totalLists / listsPerPage); i++) {
            temp.push(i);
            setPageNumbers(temp);
        }
    }, [totalLists]);

    return (
        <nav>
            <PaginationWrapper>
                {pageNumbers.map((number) => (
                    <PaginnationNumber key={number} onClick={() => setCurrentPage(number)}>
                        {number}
                    </PaginnationNumber>
                ))}
            </PaginationWrapper>
        </nav>
    );
};

Pagination.propTypes = {
    listsPerPage: PropTypes.number,
    totalLists: PropTypes.number,
    setCurrentPage: PropTypes.func,
};

export default Pagination;
