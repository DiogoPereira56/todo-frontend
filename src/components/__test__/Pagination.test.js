/* eslint-disable no-unused-vars */
/* eslint-disable no-undef */
import Pagination from '../Pagination';
import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';

const listsPerPage = 2;
const totalLists = 6;
const setCurrentPage = () => {};

describe('<Pagination />', () => {
    it('renders Pagination without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Pagination></Pagination>, div);
    });

    it('renders Pagination correctly', () => {
        const { getByTestId } = render(
            <Pagination
                listsPerPage={listsPerPage}
                totalLists={totalLists}
                setCurrentPage={setCurrentPage}
            ></Pagination>,
        );
        expect(getByTestId('page')).toHaveTextContent('123');
    });

    it('matches Pagination snapshot', () => {
        const tree = renderer.create(<Pagination></Pagination>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
