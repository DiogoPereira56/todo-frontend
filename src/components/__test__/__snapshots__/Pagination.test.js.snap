// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> matches Pagination snapshot 1`] = `
<nav>
  <div
    className="sc-fubCfw jUWFZp"
    data-testid="page"
  >
    <div
      className="sc-pFZIQ iVNzqE"
      onClick={[Function]}
    >
      1
    </div>
  </div>
</nav>
`;
